src/js/actions/search/layers.js

```diff
@@ -136,17 +136,20 @@ define(function (require, exports) {
         // Get list of layers
         var appStore = this.flux.store("application"),
             currDoc = appStore.getCurrentDocument(),
-            documents = appStore.getOpenDocuments(),
+            allLayerMaps = Immutable.List();
```

Comment:
This seems to create a bunch of unnecessary garbage Immutable objects in the process of building one final Immutable list. Can you find a way to do it that creates as little unnecessary garbage as possible? Like, for example by using a single array as an intermediate data structure and then converting it to an immutable list at the end?